### PR TITLE
[6.x/7.x-Prepatch] Scripts/Spells: Fix Shaman talent Earthen Rage

### PR DESCRIPTION
**Pull request by P-Kito**
**Changes proposed**:
- Fix Earthen Rage (http://www.wowhead.com/spell=170374)
- Shoots stones at last damaged target, as spell description says

**Target branch(es)**: 6.x/7.x-Prepatch

**Tests performed**: Working ingame

### CODE REVIEW

**Comment by P-Kito** [comment]
I broke it by removing some hacks - I might need a little help here now @Shauren 
Bascially it's not finding procTarget..?

File: src/server/scripts/Spells/spell_shaman.cpp [hunk 1]
```diff
@@ -266,6 +269,76 @@ class spell_sha_earth_shield : public SpellScriptLoader
...
+                        return false;
+                    if (!sSpellMgr->GetSpellInfo(SPELL_SHAMAN_EARTHEN_RAGE_DAMAGE))
+                        return false;
+                    return true;
+                }
+
+                void HandleEffectProc(AuraEffect const* /*aurEff*/, ProcEventInfo& eventInfo)
+                {
+                    PreventDefaultAction();
+                    _procTarget = eventInfo.GetProcTarget();
```
Comment (Ovahlord) [inline src/server/scripts/Spells/spell_shaman.cpp]: simply use  if (Unit* target = eventInfo.GetProcTarget())   and after  if (Unit* caster = eventInfo.GetActor()) so you can drop that Unit* _procTarget;  pointer.

**Comment by Shauren** [comment]
This will never work like that - what you have here are 2 separate instances of spell_sha_earthen_rage_AuraScript, belonging to 2 separate auras. Putting all this code together inside one class only adds confusion.

**Comment by P-Kito** [comment]
Talent working ingame, tested build.

**Comment by joschiwald** [comment]
what's the advantage by using an new container in aura class instead of saving data in scripts (we can access them with Aura::GetScriptByName)?

**Comment by P-Kito** [comment]
ok so obviously Spellwork didnt show the amplitude, I can remove that hack then :)

**Comment by Aokromes** [comment]
@Shauren @joschiwald this can be merged?

**Comment by Shauren** [comment]
3fedeb1c9d6b2b218f46774991bd8e841477ad1a

**Pull request closed without merging**